### PR TITLE
Improved Yoast Comatability

### PR DESCRIPTION
**Pull request by gregpriday**
This PR fixes our live Yoast integration in the classic editor. It also resolves https://github.com/siteorigin/siteorigin-panels/issues/675 and https://github.com/siteorigin/siteorigin-panels/issues/647

### CODE REVIEW

**Comment by gregpriday** [comment]
Converted to a draft while I fix some issues with the PR.

File: compat/js/siteorigin-panels-layout-block.js [hunk 1]
```diff
@@ -275,15 +278,22 @@ registerBlockType('siteorigin-panels/layout-block', {
...
         // Send panelsData to server for sanitization.
-        jQuery.post(soPanelsBlockEditorAdmin.sanitizeUrl, {
-          action: 'so_panels_layout_block_sanitize',
-          panelsData: JSON.stringify(newPanelsData)
-        }, function (sanitizedPanelsData) {
-          if (sanitizedPanelsData !== '') {
-            setAttributes({
-              panelsData: sanitizedPanelsData
-            });
+        jQuery.post(panelsOptions.ajaxurl, {
```
Comment (AlexGStapleton) [inline compat/js/siteorigin-panels-layout-block.js:281]: @gregpriday, you sent me a message about this the other day and mentioned this had changed and wasn't sure if it was correct or not. Was it correctly changed? sanitizeUrl still exists [here](https://github.com/siteorigin/siteorigin-panels/blob/feature/improved-yoast-compat/compat/layout-block.php#L59-L60).

File: inc/admin.php [hunk 2]
```diff
@@ -1059,14 +1060,57 @@ function action_builder_content() {
...
 		unset( $GLOBALS[ 'SITEORIGIN_PANELS_POST_CONTENT_RENDER' ] );
-
+		
+		wp_die();
+	}
+	
+	/**
+	 * Get builder content based on the submitted panels_data.
+	 */
+	function action_builder_content_json() {
```
Comment (AlexGStapleton) [inline inc/admin.php:1077]: This needs a nonce check just to be safe.

File: inc/admin.php [hunk 3]
```diff
@@ -1059,14 +1060,57 @@ function action_builder_content() {
 			false
 		);
 		$panels_data            = SiteOrigin_Panels_Styles_Admin::single()->sanitize_all( $panels_data );
-
+
```
Comment (AlexGStapleton) [inline inc/admin.php:1063]: Leftover tab

**Comment by AlexGStapleton** [comment]
I've tried this branch with the Block Editor and everything looks to be working well so that's awesome. Classic Editor is also fine.

Does it make sense to include the post carousel? Including potentially unrelated post titles will inflate the word count and make it partially inconsistent.

File: compat/js/siteorigin-panels-layout-block.js (hunk 1 above)
Comment (gregpriday) [reply compat/js/siteorigin-panels-layout-block.js:281]: I just had a look and `ajaxurl` does have a nonce added, so this should be fine. I imagine @braamgenis changed this to be able to hook into the action that also gives the rendered version of panels_data.

File: inc/admin.php (hunk 2 above)
Comment (gregpriday) [reply inc/admin.php:1077]: Yeah, it definitely does. Well spotted.

**Pull request merged**